frontend/packages/schema-model/src/lib/SchemaModel.ts

```diff
@@ -432,9 +438,10 @@ export class SchemaModel {
   }
 
   public deleteNode(pointer: string): SchemaModel {
-    if (pointer === ROOT_POINTER) throw new Error('It is not possible to delete the root node.');
-    if (this.isDefinitionInUse(pointer))
```

Comment:
Hva skjer nå hvis man kaller deleteNode på en type som er i bruk? Kunne det vært bedre å bytte ut denne med hasReferringNodes i stedet for å fjerne den helt?

Reply:
> Hva skjer nå hvis man kaller deleteNode på en type som er i bruk? Kunne det vært bedre å bytte ut denne med hasReferringNodes i stedet for å fjerne den helt?

Godt forslag. Jeg gjør et forsøk på det på mandag. :)

Reply:
Det skal ikke være mulig. `hasReferringNodes` sjekker kun elementet som det pekes på, mens `isDefinitionInUse` er rekursiv nettopp fordi den også sjekker oppover i treet.